### ISSUE TITLE
Site Editor Loading Error - cannot destructure 'id' of null

### ISSUE THREAD

**Issue by Addison-Stavlo** [Automattic/wp-calypso#56512]
FSE sites are encountering the following error on load of a page using a default template, resulting in a blank/gray editor.

![image](https://user-images.githubusercontent.com/28742426/134556150-4f0f20ef-4488-40fc-9418-6f656ebf72fc.png)

![image](https://user-images.githubusercontent.com/28742426/134556165-d6155f5e-a837-4640-9857-3920cb6c28f1.png)

### Initial investegation:

Looking at the error, the null destructuring seems to come from [this LOC](https://github.com/WordPress/gutenberg/blob/c8210761ff28e7b3c275e1704ad3ae1917364b2f/packages/edit-site/src/store/actions.js#L173).  The null value is returned from the `__experimentalGetTemplateForLink` selector.  

Looking further, it seems the `experimentalGetTEmplateForLink` selector is failing [here](https://github.com/WordPress/gutenberg/blob/c8210761ff28e7b3c275e1704ad3ae1917364b2f/packages/core-data/src/resolvers.js#L390).  When trying to run through this process we see that `res.json()` throws and error, resulting in the functions null return a few lines down.

![image](https://user-images.githubusercontent.com/28742426/134557038-ecd33b7a-9b4b-4d61-a570-fe1803e8c57f.png)

Sites that load fine handle the above with no error, while our problematic sites seem to have some issue with this response.

**Comment by Addison-Stavlo** [comment]
<s> unsure if this PR may be related - https://github.com/WordPress/gutenberg/pull/34579 - as it looks like the failing `.json()` call was added as part of 11.5 </s>

EDIT - that PR seems unrelated.  I found both v11.5.1 and v11.4.1 loading fine on my sandbox (most recently updated yesterday), when I updated my sandbox just now the failure started happening there as well.  Seems there is a recent dotcom change that may be the culprit.
